### ISSUE TITLE
Peer Rewiew Lab 2

### ISSUE THREAD

**Issue by AllegraRoberto** [manuelescobar-dev/Computational-Intelligence#2]
The code is well-organized, and you've applied good practices like using comments and modularization. The strategies for the Nim game and the game loop are clear. However, there are areas for improvement:
1. While you added a basic test case at the end, consider expanding your tests, especially for the evolutionary strategy.
2. Using the negative `nim_sum` function to calculate fitness might be problematic. The optimal function uses `nim_sum = 0`, so by using the negative, you unintentionally follow the optimal strategy, rendering the evolutionary approach ineffective.
3. When mutating the population, you replace parents with children. The logic of the operation is good but It could be beneficial to move the line of code creating children into a different function to enhance readability.
In summary, your code is functional, but addressing these points would make it more versatile and readable.
